### PR TITLE
Survey - Evelina Berg

### PR DESCRIPTION
**Pull request by evelinabe**
Link to my deployed project:
https://digital-elders-survey.netlify.app/

My reflections on how this week's project turned out:
I've had too much going on around me for the past few weeks, so have not been able to perform more than minimum level minus accessibility guidelines. I hope to catch up on those, as well as the radio buttons not being checked on question 4. It's really annoying since the radio buttons on next section works and I'm struggling to find the difference between them. Would like to do some touch ups regarding minor styling and a readable summary.

### CODE REVIEW

File: code/src/Components/Form.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import FirstQuestion from './FirstQuestion';
+import SecondQuestion from './SecondQuestion';
+import ThirdQuestion from './ThirdQuestion';
+import FourthQuestion from './FourthQuestion';
+import FifthQuestion from './FifthQuestion';
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:6]: Good structure, but I would ask for more explanatory names on the questions so I know if they are text-input, radio-button etc.

File: code/src/Components/Form.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+const Form = () => {
+  const [yearOfBirth, setYearOfBirth] = useState('');
+  const [gender, setGender] = useState('');
+  const [activities, setActivities] = useState([]);
+  const [recommendation, setRecommendation] = useState('');
+  const [expectations, setExpectations] = useState('');
+  const [textInput, setTextInput] = useState('');
+  const [section, setSection] = useState('firstQuestion');
+  const [printString, setPrintString] = useState('');
+  const handleYearOfBirthChange = event => {
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:19]: Good namings! 👍🏼

File: code/src/Components/Form.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    setTextInput(event);
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    setSection("summary")
+  }
+
+
+  console.log(textInput);
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:55]: Maybe remove the console.logs when handing in the project. :)

File: code/src/Components/Form.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+
+return (
+  <>
+    <form onSubmit={handleSubmit}>
+    {section === 'firstQuestion' && (
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:61]: Indent missing!

File: code/src/Components/Form.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      <>  
+        <FirstQuestion
+        userYearofBirth={yearOfBirth}
+        onYearOfBirthChange={handleYearOfBirthChange}
+        section={section}
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:66]: Indents missing

File: code/src/Components/Form.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    {section === 'secondQuestion' && (
+      <>
+        <SecondQuestion 
+          userGender={gender}
+          onGenderChange={handleGenderChange}
+          section={section}
+          setSection={setSection}
+        />
+      </>
+    )}
```
Comment (Gabbi-89) [inline code/src/Components/Form.js:80]: Good indentions 👍🏼

File: code/src/Components/Summary.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+    return (
+        <div>
+            <p>
+                Jag är en {userGender} född år {userYearofBirth}. 
+                Jag har deltagit i aktiviteten {userActivities}. 
+                {userActivities} kan jag {userRecommendation} rekommendera
+                till en vän. Aktiviteten uppfyllde {userExpectation} mina förväntningar. 
+                Jag vill också skicka med er {userTextInput}.
+            </p>
```
Comment (Gabbi-89) [inline code/src/Components/Summary.js:13]: Recommendation is to change them to eg. several <p> tags so it gets space in the summary at the webpage. Just to read the summary easier :)

File: code/src/Components/SubmitButton.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+
+const SubmitButton = () => {
+    
+    return (
```
Comment (Gabbi-89) [inline code/src/Components/SubmitButton.js:5]: One indent to much

File: code/src/Components/FifthQuestion.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+import Button from './Button';
+
+const FifthQuestion = ({ userRecommendation, onRecommendationChange, section, setSection }) => {
+
+    return (
+      <section className="question-section">
+        <h2 className="question-header">Skulle du rekommendera aktiviteten till en vän?</h2>
+        <div className="radio-buttons recommendation">
+          <label htmlFor="redanGjort">
+          <input
```
Comment (Gabbi-89) [inline code/src/Components/FifthQuestion.js:11]: Miss on indention

File: code/src/Components/FourthQuestion.js [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+          <label htmlFor="jajjemän">
+          <input 
+            type="radio"
+            name="expect"
+            value="jajjemän"         
+            checked={userExpectations === "jajjemän"}  
+            onChange={onExpectationsChange} 
+            printString="jajjemän"   
+            className="radio-button"
+          />
```
Comment (Gabbi-89) [inline code/src/Components/FourthQuestion.js:19]: Missing indention

File: code/src/Components/FourthQuestion.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import Button from './Button.js'
+
+const FourthQuestion = ({ userExpectations, onExpectationsChange, section, setSection, printString, setPrintString }) => {
```
Comment (Gabbi-89) [inline code/src/Components/FourthQuestion.js:4]: I tried to understand why this one isn't working but the fifth question with radiobuttons is working..
The difference that I found is `printString` and `setPrintString` - these are not included in the fifth question. Try to remove them to see if that works :)

File: code/src/Components/FourthQuestion.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+        <h2 className="question-header">Uppfyllde aktiviteten dina förväntningar?</h2>
+        <div className="radio-buttons expectation">
+          <label htmlFor="jajjemän">
+          <input 
+            type="radio"
+            name="expect"
+            value="jajjemän"         
+            checked={userExpectations === "jajjemän"}  
+            onChange={onExpectationsChange} 
+            printString="jajjemän"
```
Comment (Gabbi-89) [inline code/src/Components/FourthQuestion.js:17]: Also try to remove printString in the <input> labels :)

File: code/src/Components/SixthQuestion.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+        
+        <label htmlFor="userTextInput">Skriv här:</label>
+        <input 
+          value={userTextInput}
+          onChange={e => onTextInputChange(e.target.value)}
+          id="userTextInput"
+          type="textarea"
+          placeholder="Det var roligt att..."
+          required
+        />
```
Comment (Gabbi-89) [inline code/src/Components/SixthQuestion.js:18]: Missing indention

File: code/src/Components/SixthQuestion.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        <SubmitButton 
+            disable={onTextInputChange}
```
Comment (Gabbi-89) [inline code/src/Components/SixthQuestion.js:20]: Good!!!

File: code/src/Components/ThirdQuestion.js [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+                <label htmlFor="rådAvCoach">Råd av en coach</label>
+                <input
+                    id="rådAvCoach"
+                    type="checkbox"
+                    checked={userActivities.includes('rådAvCoach')}
+                    onChange={() => onActivitiesChange('rådAvCoach')}
+                />
+            </div>
+            <Button 
+                disable={onActivitiesChange}
```
Comment (Gabbi-89) [inline code/src/Components/ThirdQuestion.js:41]: Disable function doesn't work. Try to set 
`disable={userActivities === '' }`

File: code/src/Components/FirstQuestion.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        onChange={onYearOfBirthChange}
+        id="yearOfBirth"
+        type="number"
+        min="1900"
+        max="2020"
+        placeholder="År"
+        required
+      />
+      <Button 
+        disable={onYearOfBirthChange === false}
```
Comment (Gabbi-89) [inline code/src/Components/FirstQuestion.js:21]: Disable functions doesn't work, try to se {userYearOfBirth === ""}

File: code/src/Components/SecondQuestion.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+        value={userGender}
+        onChange={e => onGenderChange(e)}
+      >
+        <option value="">Välj</option>
+        <option value="kvinna">Kvinna</option>
+        <option value="man">Man</option>
+        <option value="annat">Annat</option>
+      </select>
+      <Button 
+        disable={onGenderChange}
```
Comment (Gabbi-89) [inline code/src/Components/SecondQuestion.js:19]: Disablebutton doesn't work, try to set {userGender === ""}

**Review by Gabbi-89 — CHANGES_REQUESTED** [review]
Evelina, you are very great in writing good, correct and nice code! It is clear structure of the code and I do understand everything. You have built up the form with a good component-structure! The theme of the form is also very good! 

I would just want you to fix some of the indentions and extra spaces and to fix the fourth question that doesn't work right now. I gave you a tip for how to solve it. If it doesn't work, please let me know and I will try to help you.

I did also gave you a tip for how to make the disable buttons to work. Try that, otherwise you can remove the 'disabled'.

You have really succeeded with the task despite that you have had a really tough period privately. Please reach out to me if I shall help you solve anything!

File: code/src/Components/SecondQuestion.js (hunk 17 above)
Comment (evelinabe) [reply code/src/Components/SecondQuestion.js]: Couldn't get it to work, took another route :-)

File: code/src/Components/FourthQuestion.js (hunk 11 above)
Comment (evelinabe) [reply code/src/Components/FourthQuestion.js]: They work now, not really sure what I did. 
It was not the `printString` though. That was an effort by me to try to print the value out in normal text in the summary, but I couldn't get that to work either 🙈. Removed them for now anyhow.

**Review by Gabbi-89 — APPROVED** [review]
So nice that you solved the radio buttons in question four! 💯  Extra plus for making the inputs required 👍🏼

File: code/src/App.js [hunk 18]
```diff
@@ -1,9 +1,9 @@
...
 
 export const App = () => {
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
-  )
+      <Form />
+  );
 }
```
Comment (HIPPIEKICK) [inline code/src/App.js:9]: Semicolon

File: code/src/Components/Button.js [hunk 19]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+
+const Button = ({ defaultState, currentState, disabled, setSection, nextQuestion }) => {
+  const handleButton = event => {
+    if(currentState !== defaultState) { 
+      console.log(currentState);
+      console.log(defaultState);
+      setSection(nextQuestion);
+    }
+}
```
Comment (HIPPIEKICK) [inline code/src/Components/Button.js:10]: semicolon

File: code/src/Components/ThirdQuestion.js [hunk 20]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import Button from "./Button";
+
+const ThirdQuestion = ({
```
Comment (HIPPIEKICK) [inline code/src/Components/ThirdQuestion.js:4]: These checkboxes are acting as radio buttons right now and the summary is not showing any of this input. Take another look at Maks' lecture/code about it -> https://technigo.slack.com/archives/C017W9LGX3R/p1603279217488000

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
The usage of semicolons is inconsistent, so have another look at your files to make it consistent. Also, the third question is not working properly, sent you a link with Maks' lecture that will hopefully help you with that :)

File: code/src/Components/ThirdQuestion.js (hunk 20 above)
Comment (evelinabe) [reply code/src/Components/ThirdQuestion.js]: This was a tough one! Had a 1:1 with Max with no solution in the 20 minutes we had. Discovered after that it was a line in the button that had messed it up. Now it works as intended 😅

Comment (HIPPIEKICK) [reply code/src/Components/ThirdQuestion.js]: Great!

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**